src/App.tsx

```diff
@@ -1,10 +1,69 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Notification } from './components/Notification/Notification';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+
+const USER_ID = 10140;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+
+  const [status, setStatus] = useState<string>('all');
+  const [itemsLeft, setItemsLeft] = useState<number>(0);
+  const [itemsCompleted, setitemsCompleted] = useState<number>(0);
+
+  const countItemsLeft = todos.filter(todo => !todo.completed).length;
+  const countItemsCompleted = todos.filter(todo => todo.completed).length;
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((fetchedTodos: Todo[]) => {
+        setTodos(fetchedTodos);
+      })
+      .catch((fetchedError: Error) => {
+        setError(fetchedError?.message ?? 'Something went wrong');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if (error) {
+      timeout = setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [error]);
+
+  const filtredTodos = useMemo(() => {
+    switch (status) {
+      case 'all':
+        return todos;
+      case 'active':
+        return todos.filter((todo) => !todo.completed);
+      case 'completed':
+        return todos.filter((todo) => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, status]);
+
+  useEffect(() => {
+    setItemsLeft(countItemsLeft);
+    setitemsCompleted(countItemsCompleted);
+  }, [status, todos]);
```

Comment:
The useEffect should have 'countItemsLeft' and 'countItemsCompleted' as dependencies instead of 'status' and 'todos'

Reply:
or you can do the same with these vars and use them as vars, not as useState